game/src/graphic/textures/TextureHandler.java

```diff
@@ -2,30 +2,70 @@
 
 import com.badlogic.gdx.Gdx;
 import com.badlogic.gdx.files.FileHandle;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.Arrays;
+import java.util.Comparator;
 import java.util.LinkedHashMap;
 import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
 import java.util.Set;
+import java.util.function.Predicate;
+import java.util.logging.Logger;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 /**
  * A texture handler class for managing textures paths for further use.
  *
- * <p>Singleton (possibly thread-safe).
+ * <p>This class is designed as Singleton, because all asset paths should be read only once at the
+ * beginning of the application.
  */
 public class TextureHandler {
+    public static final String PLACEHOLDER_FILENAME = ".resource_root";
+
     private static final TextureHandler INSTANCE = new TextureHandler();
 
     private final Map<String, Set<FileHandle>> pathMap = new LinkedHashMap<>();
 
     private TextureHandler() {
-        addAllAssets(Gdx.files.internal(Gdx.files.getLocalStoragePath()));
+        addAllAssets(Objects.requireNonNull(getResourceRoot()));
+    }
+
+    /**
+     * Searches for the resource root that has the longest path name string. Internal helper method.
```

Comment:
Hier fehlt noch komplett das Konzept, was Du Dir überlegt hast. Was macht diese Methode, wie soll sie ihr Ergebnis erreichen? Warum die "4" weiter unten? Was sind die gemachten Annahmen, wie muss das Projektlayout aussehen? 

Hatte ich bereits in einem der früheren Reviews schon mal geschrieben - ich habe keine Lust, das in jedem einzelnen Review immer wieder aufschreiben zu müssen :/

Reply:
"Internal helper method." kann weg - das Ding ist `private`, das bedeutet "interne Hilfsmethode" :)